src/explore-education-statistics-admin/src/pages/release/datablocks/ReleaseDataBlocksPage.tsx

```diff
@@ -36,41 +34,70 @@ const ReleaseDataBlocksPage = ({
     ReleaseDataBlockSummary
   >();
 
-  const { value: model, isLoading, setState: setModel } = useAsyncHandledRetry<
-    Model
-  >(async () => {
-    const [dataBlocks, canUpdateRelease] = await Promise.all([
-      dataBlocksService.listDataBlocks(releaseId),
-      permissionService.canUpdateRelease(releaseId),
-    ]);
+  const queryClient = useQueryClient();
+
+  const listFeaturedTablesQuery = useMemo(
+    () => featuredTableQueries.list(releaseId),
+    [releaseId],
+  );
+  const listDataBlocksQuery = useMemo(() => dataBlockQueries.list(releaseId), [
+    releaseId,
+  ]);
 
-    return {
-      dataBlocks,
-      canUpdateRelease,
-    };
-  }, [releaseId]);
+  const { data: dataBlocks = [], isLoading: isLoadingDataBlocks } = useQuery(
+    listDataBlocksQuery,
+  );
+  const {
+    data: featuredTables = [],
+    isLoading: isLoadingFeaturedTables,
+  } = useQuery(listFeaturedTablesQuery);
+  const {
+    data: canUpdateRelease = true,
+    isLoading: isLoadingPermissions,
+  } = useQuery(permissionQueries.canUpdateRelease(releaseId));
 
   const handleDelete = useCallback(async () => {
-    if (!deleteDataBlock || !model) {
+    if (!deleteDataBlock) {
       return;
     }
 
-    setModel({
-      value: {
-        ...model,
-        dataBlocks: model.dataBlocks.filter(
-          dataBlock => dataBlock.id !== deleteDataBlock.id,
-        ),
-      },
-    });
+    queryClient.setQueryData(
+      listDataBlocksQuery.queryKey,
+      dataBlocks.filter(dataBlock => dataBlock.id !== deleteDataBlock.id),
+    );
+
+    queryClient.setQueryData(
+      listFeaturedTablesQuery.queryKey,
+      featuredTables.filter(table => table.dataBlockId !== deleteDataBlock.id),
+    );
+    await queryClient.invalidateQueries(listDataBlocksQuery.queryKey);
```

Comment:
Just wondering why we don't pass both query keys in one go since `invalidateQueries` accepts an array of keys to invalidate?

i.e. 
```
    await queryClient.invalidateQueries([listDataBlocksQuery.queryKey, listFeaturedTablesQuery.queryKey]);
```

Reply:
Because I wasn't thinking :) I've sorted it now.